src/nspawn/nspawn-util.c

```diff
@@ -49,6 +49,7 @@ int systemd_installation_has_version(const char *root, unsigned minimal_version)
                         /* This is most likely to run only once, hence let's not optimize anything. */
                         char *t, *t2;
                         unsigned version;
+                        _cleanup_free_ char *string = NULL;
```

Comment:
quite frankly I think we should drop the numeric version comparison altogether. instead let's just use strverscmp_improved() for this stuff. We have it for stuff like that, and it should handle this in a much nicer way.

Reply:
strverscmp_improved() works great.